src/main/java/ca/team2706/frc/robot/commands/drivebase/DriverAssistVision.java

```diff
@@ -311,99 +444,162 @@ public void generateTrajectoryRobotToTarget(double distanceCameraToTarget_Camera
-
-        // Compute vector from target to final robot position in field frame from unit vector in field frame
-        double d = Config.TARGET_OFFSET_DISTANCE.value();
-        double vTargetToFinal_FieldX = -d * vUnitFacingTarget_FieldX;
-        double vTargetToFinal_FieldY = -d * vUnitFacingTarget_FieldY;
-        Log.d("DAV: vec_TargetToFinalX_Field: " + vTargetToFinal_FieldX + ", vTargetToFinal_FieldY: " + vTargetToFinal_FieldY);
+        // Vector 1+2: vRobotToTarget_Robot: Vector from robot to target in robot frame
+        double vRobotToTarget_RobotX = vRobotToCamera_RobotX + vCameraToTarget_RobotX;
+        double vRobotToTarget_RobotY = vRobotToCamera_RobotY + vCameraToTarget_RobotY;
 
         // Vector 3: vTargetToFinal_Robot: Vector from target to final robot position in robot frame
-        //           (need to do a coordinate frame transformation on vTargetToFinal_Field)
-        // Note: angRobotCurrent_Field is the current angle from the x axis of robot frame with respect to the
-        //       x axis of the field frame, and the x axis of the robot frame is 90deg less than the robot heading
-        //       which points along the y axis by definition
-        double angRobotCurrent_Field = angRobotHeadingCurrent_Field - 90.0;
-        double angRobotCurrentRad_Field = Pathfinder.d2r(angRobotCurrent_Field);
-        double cosAngRobotCurrentRad_Field = Math.cos(angRobotCurrentRad_Field);
-        double sinAngRobotCurrentRad_Field = Math.sin(angRobotCurrentRad_Field);
-        double vTargetToFinal_RobotX = vTargetToFinal_FieldX * cosAngRobotCurrentRad_Field + vTargetToFinal_FieldY * sinAngRobotCurrentRad_Field;
-        double vTargetToFinal_RobotY = -vTargetToFinal_FieldX * sinAngRobotCurrentRad_Field + vTargetToFinal_FieldY * cosAngRobotCurrentRad_Field;
-        Log.d("DAV: vTargetToFinal_RobotX: " + vTargetToFinal_RobotX + ", vTargetToFinal_RobotY: " + vTargetToFinal_RobotY);
-
-        // Compute vRobotToFinal_Robot as sum of Vectors 1, 2, and 3
-        // vRobotToFinal_Robot = vRobotToCamera_Robot + vCameraToTarget_Robot + vTargetToFinal_Robot
-        double vRobotToFinal_RobotX = vRobotToCamera_RobotX + vCameraToTarget_RobotX + vTargetToFinal_RobotX;
-        double vRobotToFinal_RobotY = vRobotToCamera_RobotY + vCameraToTarget_RobotY + vTargetToFinal_RobotY;
+        // Depends on target type.
+        double vTargetToFinal_RobotX = 0.0;
+        double vTargetToFinal_RobotY = 0.0;
+        double angRobotCurrent_Field = 0.0;
+        if ((target == DriverAssistVisionTarget.CARGO_AND_LOADING) ||
+            (target == DriverAssistVisionTarget.ROCKET)                ) {
+            // Vector 3 is aligned with and facing away from target
+            // Get current robot heading relative to field frame from IMU
+            double angRobotHeadingCurrent_Field = DriveBase.getInstance().getAbsoluteHeading() % 360;
+            if(angRobotCurrent_Field < 0) {
```

Comment:
I don't recall having put this code there. It may have been an "auto-completion error". In any case, angRobotCurrent_Field in the condition and body should actually be angRobotHeadingCurrent_Field. angRobotCurrent_Field is defined in a comment a little later on where it is first used. This has been fixed and pushed to github.